arm-operationalinsights/2015-11-01-preview/swagger/OperationalInsights.json

```diff
@@ -0,0 +1,1230 @@
+        }
+      },
+      "description": "A management group that is connected to a workspace"
+    },
+    "WorkspaceListManagementGroupsResult": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManagementGroup"
+          },
+          "description": "Gets or sets a list of management groups attached to the workspace."
+        }
+      },
+      "description": "The list workspace managmement groups operation response."
+    },
+    "Sku": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the SKU.",
+          "enum": [
+            "Free",
+            "Standard",
+            "Premium",
+            "Unlimited",
+            "PerNode",
+            "Standalone"
+          ],
+          "x-ms-enum": {
+            "name": "SkuNameEnum",
+            "modelAsString": true
+          }
+        }
+      },
+      "required": [
+        "name"
+      ],
+      "description": "The SKU (tier) of a workspace."
+    },
+    "WorkspaceProperties": {
+      "properties": {
+        "provisioningState": {
+          "type": "string",
+          "description": "The provisioning state of the workspace.",
+          "enum": [
+            "Creating",
+            "Succeeded",
+            "Failed",
+            "Canceled",
+            "Deleting",
+            "ProvisioningAccount"
+          ],
+          "x-ms-enum": {
+            "name": "EntityStatus",
+            "modelAsString": true
+          }
+        },
+        "source": {
+          "type": "string",
+          "description": "The source of the workspace.  Source defines where the workspace was created. 'Azure' implies it was created in Azure.  'External' implies it was created via the Operational Insights Portal. This value is set on the service side and read-only on the client side."
+        },
+        "customerId": {
+          "type": "string",
+          "description": "The ID associated with the workspace.  Setting this value at creation time allows the workspace being created to be linked to an existing workspace."
+        },
+        "portalUrl": {
+          "type": "string",
+          "description": "The URL of the Operational Insights portal for this workspace.  This value is set on the service side and read-only on the client side."
+        },
+        "sku": {
+          "$ref": "#/definitions/Sku",
+          "description": "The SKU of the workspace."
+        },
+        "retentionInDays":{
+          "type":"integer",
+          "format": "int32",
+          "minimum": -1,
+          "maximum": 730,
+          "description": "The workspace data retention in days. -1 means Unlimited retention for the Unlimited Sku. 730 days is the maximum allowed for all other Skus. "
+        }
+      },
+      "description": "Workspace properties."
+    },
+    "Workspace": {
+      "properties": {
```

Comment:
You use this in `CreateOrUpdate...`, can any of those properties (i.e. also applies to `WorkspaceProperties`) be marked as required?